pg_shard.c

```diff
@@ -937,6 +973,12 @@ BuildDistributedPlan(Query *query, List *shardIntervalList)
 
 		deparse_shard_query(query, shardId, queryString);
 
+		if (LogDistributedStatements)
+		{
+			ereport(LOG, (errmsg("distributed statement: %s", queryString->data),
+						  errhidestmt(true)));
```

Comment:
The comment for errhidestmt mentions that it can be used for error messages which contain the statement. This one doesn't contain the original statement (it contains the transformed query that we send down to the worker nodes). Is it useful to retain that, so we can tie which transformed statement was a result of which original user query?